v2/api/insights/v1api20230311/structure.txt

```diff
@@ -107,6 +107,467 @@ DataCollectionEndpoint: Resource
+│   │   │   └── Streams: Enum (1 value)[]
+│   │   │       └── "Microsoft-PrometheusMetrics"
+│   │   ├── Syslog: Object (5 properties)[]
+│   │   │   ├── FacilityNames: Enum (27 values)[]
+│   │   │   │   ├── "alert"
+│   │   │   │   ├── "audit"
+│   │   │   │   ├── "auth"
+│   │   │   │   ├── "authpriv"
+│   │   │   │   ├── "clock"
+│   │   │   │   ├── "cron"
+│   │   │   │   ├── "daemon"
+│   │   │   │   ├── "ftp"
+│   │   │   │   ├── "kern"
+│   │   │   │   ├── "local0"
+│   │   │   │   ├── "local1"
+│   │   │   │   ├── "local2"
+│   │   │   │   ├── "local3"
+│   │   │   │   ├── "local4"
+│   │   │   │   ├── "local5"
+│   │   │   │   ├── "local6"
+│   │   │   │   ├── "local7"
+│   │   │   │   ├── "lpr"
+│   │   │   │   ├── "mail"
+│   │   │   │   ├── "mark"
+│   │   │   │   ├── "news"
+│   │   │   │   ├── "nopri"
+│   │   │   │   ├── "ntp"
+│   │   │   │   ├── "*"
+│   │   │   │   ├── "syslog"
+│   │   │   │   ├── "user"
+│   │   │   │   └── "uucp"
+│   │   │   ├── LogLevels: Enum (9 values)[]
+│   │   │   │   ├── "Alert"
+│   │   │   │   ├── "Critical"
+│   │   │   │   ├── "Debug"
+│   │   │   │   ├── "Emergency"
+│   │   │   │   ├── "Error"
+│   │   │   │   ├── "Info"
+│   │   │   │   ├── "Notice"
+│   │   │   │   ├── "*"
+│   │   │   │   └── "Warning"
+│   │   │   ├── Name: *string
+│   │   │   ├── Streams: Enum (1 value)[]
+│   │   │   │   └── "Microsoft-Syslog"
+│   │   │   └── TransformKql: *string
+│   │   ├── WindowsEventLogs: Object (4 properties)[]
+│   │   │   ├── Name: *string
+│   │   │   ├── Streams: Enum (2 values)[]
+│   │   │   │   ├── "Microsoft-Event"
+│   │   │   │   └── "Microsoft-WindowsEvent"
+│   │   │   ├── TransformKql: *string
+│   │   │   └── XPathQueries: string[]
+│   │   └── WindowsFirewallLogs: Object (3 properties)[]
+│   │       ├── Name: *string
+│   │       ├── ProfileFilter: Enum (3 values)[]
+│   │       │   ├── "Domain"
+│   │       │   ├── "Private"
+│   │       │   └── "Public"
+│   │       └── Streams: string[]
+│   ├── Description: *string
+│   ├── Destinations: *Object (10 properties)
+│   │   ├── AzureDataExplorer: Object (3 properties)[]
+│   │   │   ├── DatabaseName: *string
+│   │   │   ├── Name: *string
+│   │   │   └── ResourceReference: *genruntime.ResourceReference
+│   │   ├── AzureMonitorMetrics: *Object (1 property)
+│   │   │   └── Name: *string
+│   │   ├── EventHubs: Object (2 properties)[]
+│   │   │   ├── EventHubResourceReference: *genruntime.ResourceReference
+│   │   │   └── Name: *string
+│   │   ├── EventHubsDirect: Object (2 properties)[]
+│   │   │   ├── EventHubResourceReference: *genruntime.ResourceReference
+│   │   │   └── Name: *string
+│   │   ├── LogAnalytics: Object (2 properties)[]
+│   │   │   ├── Name: *string
+│   │   │   └── WorkspaceResourceReference: *genruntime.ResourceReference
+│   │   ├── MicrosoftFabric: Object (5 properties)[]
+│   │   │   ├── ArtifactId: *string
+│   │   │   ├── DatabaseName: *string
+│   │   │   ├── IngestionUri: *string
```

Comment:
There are some fields in here like `IngestionUri` which could support `FromConfig` options. I don't know how crazy we want to go there though... do you think we should leave this as-is or add FromConfig for some of these?